lib/internal/modules/helpers.js

```diff
@@ -353,6 +353,15 @@ function shouldRetryAsESM(errorMessage, source) {
   return false;
 }
 
+/**
+ * @param {string|undefined} url URL to convert to filename
+ */
+function urlToFilename(url) {
+  if (url && StringPrototypeStartsWith(url, 'file://')) {
+    return fileURLToPath(url);
+  }
+  return url;
+}
```

Comment:
We could use that util in more places: https://github.com/nodejs/node/blob/d6b57f6629da5616362be60ae0f3e9510a7f700e/lib/internal/modules/esm/translators.js#L244 https://github.com/nodejs/node/blob/d6b57f6629da5616362be60ae0f3e9510a7f700e/lib/internal/modules/esm/translators.js#L266